include/plat/pc99/plat/64/plat_mode/machine/hardware.bf

```diff
@@ -42,4 +42,23 @@ block vtd_pte {
     field       read                1
 }
 
+-- Intel VT-d Interrupt Remapping Table Entry for Remapped Interrupts
+block vtd_irte {
+    padding                         44
+    field       sid_vtype           2
+    field       sid_q               2
+    field       source_id           16
+    field       dest_id             32
+    padding                         8
+    field       vector              8
+    field       irte_mode           1
+    padding                         7
+    field       delivery_mode       3
+    field       trigger_mode        1
+    field       redir_hint          1
+    field       dest_mode           1
+    field       fault_disable       1
+    field       present             1
+}
+
 #endif
```

Comment:
This seems the same as the 32-bit one, perhaps put one version in include/arch/x86/arch/object/structures.bf instead?

Reply:
I am not really familiar with the kernel. Does `structures.bf` and `hardware.bf` work the same way? The [docs](https://github.com/seL4/seL4/blob/61a8ccef44eeb08ecd20ad40296340c8319adee5/tools/bitfield_gen.md?plain=1#L788C39-L788C47) doesn't say much about it.

Reply:
I'm not very familiar with bitfield generation either, but I think it should just work?